internal/config/config_test.go

```diff
@@ -878,3 +880,45 @@ func getConfigAsMap(conf *Config, t *testing.T) (res configMap) {
 	}
 	return
 }
+
+type dummyConfigAPIHandler struct{}
```

Comment:
```suggestion
type fakeConfigAPIHandler struct{}
```
nit: naming

Reply:
you really dislike dummies ;)  In the testing community there is a distinction between mocks, fakes, dummies, spies.
A fake implies some kind of logic whereas a dummy just returns canned responses.  
https://blog.cleancoder.com/uncle-bob/2014/05/14/TheLittleMocker.html

Why do you want to use fake?

Reply:
As you already said yourself, a fake implies some kind of logic, which is the case for this handler, no?

The distinction between dummies, fakes, mocks, or dummies is, imo, most of the time rather unclear and a lot of those terms get used interchangeably by people (which also gets touched on in the article you linked).
Going strictly by the terminology laid out by the article the api handler would either be a stub, or a fake, since we *do* care about how it is used, and there is a slight bit of logic in the handler